### PR TITLE
feat(datasetDownloadButton): implement "Download survey data" button

### PR DESCRIPTION
**Pull request by jalMogo**
Resolves: https://github.com/jalMogo/mgmt/issues/26

TODO's:
 - [x] depends on merging: https://github.com/mapseed/api/pull/140
 - [x] refactor references to the old AuthNavView in the flavor-specific AppView's

### CODE REVIEW

File: src/base/static/js/views/app-view.js [hunk 1]
```diff
@@ -221,6 +224,23 @@ module.exports = Backbone.View.extend({
...
+    const datasetOwner = storeState.config.app.dataset_download.owner;
+    const apiRoot = storeState.config.app.api_root;
+    const isAdmin = Util.getAdminStatus(datasetSlug);
+    if (isAdmin) {
+      ReactDOM.render(
+        <Link
+          href={`${apiRoot}${datasetOwner}/datasets/${datasetSlug}/mapseed-places.csv?format=csv&include_private=true&page_size=10000`}
+          style={{ float: "right" }}
+        >
+          <SecondaryButton>{`Download Survey Data`}</SecondaryButton>
```
Comment (goldpbear) [inline src/base/static/js/views/app-view.js]: I don't think the nested `SecondaryButton` is needed here-- I think the `Link` component by itself should be enough.

Comment (jalMogo) [reply src/base/static/js/views/app-view.js]: I suppose a link looks a bit cleaner. This is how it currently looks with just the link:
download-data-header

in re:
>An alternative UI might nest the download button in the flyout that appears when you click on a logged-in user avatar. 

I think that is preferable. Downloading the dataset is an infrequently used feature, so it only clutters the limited mobile space by keeping it up there.

I may need to port the PagesNavView view to React, so I'll explore whether that is worth doing.

File: src/base/static/components/molecules/user-menu.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+              src={this.props.currentUser.avatar_url}
+            />
+          </a>
+          <ul
+            className={classNames("menu", "sign-in-menu", {
+              "is-exposed": this.state.isMenuOpen,
+            })}
+          >
+            {isAdmin && (
+              <li style={{ paddingBottom: "16px" }}>
```
Comment (jalMogo) [inline src/base/static/components/molecules/user-menu.js]: I'm not too excited about the inline styles here and the confusing state of our CSS.

After this PR is merged, I think I'll take a pass on this component using CSS-in-JS and see if it cleans things up.

File: src/base/static/js/views/pages-nav-view.js [hunk 3]
```diff
@@ -50,8 +49,4 @@ module.exports = Backbone.View.extend({
     );
   },
 
-  onAuthNavClick: function(evt) {
-    evt.preventDefault();
-    $(".sign-in-menu").toggleClass("is-exposed");
-  },
```
Comment (jalMogo) [inline src/base/static/js/views/pages-nav-view.js]: This code was not being used - it seems irrelevant to PagesNavView, because it is specific to AuthNavView. Perhaps it was left over from a copy/paste operation?

Either way, it was confusing and unused, so I removed it.

File: webpack.config.js [hunk 4]
```diff
@@ -156,5 +156,6 @@ module.exports = {
     },
     compress: true,
     port: PORT,
+    allowedHosts: [".ngrok.io"],
```
Comment (jalMogo) [inline webpack.config.js]: This was needed for testing with ngrok.

File: src/base/static/components/molecules/user-menu.js (hunk 2 above)
Comment (goldpbear) [reply src/base/static/components/molecules/user-menu.js]: Do we want to move the CSS used here into a `user-menu.scss` file and apply BEM styles? That's how we've handled other components ports so far.

File: src/base/static/components/molecules/user-menu.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                <Link
+                  href={`${
+                    this.props.apiRoot
+                  }${datasetOwner}/datasets/${datasetSlug}/mapseed-places.csv?format=csv&include_private=true&page_size=10000`}
+                  classes={".btn-secondary"}
+                  style={{ width: "100%", textDecoration: "none" }}
```
Comment (goldpbear) [inline src/base/static/components/molecules/user-menu.js]: Ideally we shouldn't have to pass through legacy CSS classes like `btn-secondary`. The `Link` component should have data-driven styles in it that replicate the styling in the legacy scss files.

I think if you pass a `color="secondary"` prop to a `Link` it will adopt the style of the Sign-In button.

File: src/base/static/components/molecules/user-menu.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      return {
+        ...prevState,
+        isMenuOpen: !prevState.isMenuOpen,
+      };
+    });
+  };
+
+  render() {
+    const {
+      slug: datasetSlug,
```
Comment (goldpbear) [inline src/base/static/components/molecules/user-menu.js]: If I run this code on a flavor without a `dataset_download` section in the config, I get an error:
```
user-menu.js:82 Uncaught TypeError: Cannot read property 'slug' of undefined
```
Maybe we should have a default value for the `datasetDownloadConfig` of an empty object.

**Comment by jalMogo** [comment]
@goldpbear I've made the following updates:
 - fixed the bug when a `dataset_download` config is not present on the flavor
 - updated august, bathgate, hull, manzanares, and williams flavors to reference the new UserMenu component. All references to AuthNavView should now be gone.

I'm not planning to make any of the CSS changes in the PR - I am working on a CSS-in-JS implementation that will clean up the CSS in UserMenu, as well as add support for theming. If that's alright with you, maybe we can merge this and I'll submit a followup PR within a couple hours.

**Comment by goldpbear** [comment]
Sounds good-- I think it's fine to wait on the CSS changes.

I'll go ahead and close https://github.com/mapseed/platform/pull/992 since it looks like the VV flavors got addressed in this PR.
